### PR TITLE
Create Initial Tailwind Preset Package with colors

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**

This PR introduces the logic for the `@metamask/design-system-tailwind-preset` to the MetaMask Design System monorepo. This package provides a Tailwind CSS preset that encapsulates MetaMask's design tokens and styling conventions for color making it easier for developers to create consistent UIs across MetaMask Tailwind projects. Subsequent PRs will add shadow, typography, space, etc

**NOTE:** This PR does not contain tests. Tests are in a separate PR, as they need to check against the design tokens stylesheet and are quite intensive, deserving their own PR and review: https://github.com/MetaMask/metamask-design-system/pull/51

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-design-system/issues/34

## **Manual testing steps**

1. Pull this branch.
2. Navigate to the `design-system-tailwind-preset` package: `cd package/design-system-tailwind-preset`.
3. Run `yarn build`.
4. Ensure you're using `yarn version 1.22.22`: `yarn set version 1.22.22`.
5. Run `yarn link`.
6. Navigate to the `metamask-portfolio` repo.
7. Pull this branch: [feat/design-system-tailwind-preset](https://github.com/consensys-vertical-apps/metamask-portfolio/tree/feat/design-system-tailwind-preset).
8. Run `yarn link "@metamask/design-system-tailwind-preset"`.
9. Run `yarn && yarn start`.
10. Verify that the MetaMask design tokens are available as Tailwind classes.

## **Screenshots/Recordings**

The recording shows a locally linked `@metamask/design-system-tailwind-preset` package inside the `metamask-portfolio`. The `tailwind.config.js` has been updated, and the design token-generated Tailwind class names have been removed in favor of the preset package. The preset package is now responsible for generating all of the design token-aligned Tailwind class names for color.

https://github.com/user-attachments/assets/f13350cb-e64c-44d1-b7d5-020cff07b011

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs)
- [x] I've completed the PR template to the best of my ability
- [ ] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by georgewrmarshall** [comment]
@SocketSecurity ignore-all

File: packages/design-system-tailwind-preset/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# `@metamask/design-system-tailwind-preset`
+
+Design system Tailwind CSS preset for MetaMask projects
```
Comment (georgewrmarshall) [inline packages/design-system-tailwind-preset/README.md:3]:
```suggestion
Design System Tailwind CSS preset for MetaMask projects
```

Make sure to update in `package.json` as well

File: packages/design-system-tailwind-preset/package.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
+{
+  "name": "@metamask/design-system-tailwind-preset",
+  "version": "0.0.0",
+  "description": "Design system Tailwind CSS preset for MetaMask projects",
```
Comment (georgewrmarshall) [inline packages/design-system-tailwind-preset/package.json:4]:
```suggestion
  "description": "Design System Tailwind CSS preset for MetaMask projects",
```

File: packages/design-system-tailwind-preset/src/colors.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
+export const colors = {
```
Comment (georgewrmarshall) [inline packages/design-system-tailwind-preset/src/colors.ts:1]: Break out colors into separate PR

File: packages/design-system-tailwind-preset/src/index.test.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import tailwindConfig from '.';
```
Comment (georgewrmarshall) [inline packages/design-system-tailwind-preset/src/index.test.ts:1]: Improve tests

File: packages/design-system-tailwind-preset/README.md (hunk 1 above)
Comment (georgewrmarshall) [reply packages/design-system-tailwind-preset/README.md]: Updated https://github.com/MetaMask/metamask-design-system/pull/41/commits/9e08c779c7f9cc64c0ed7054d62d92631a661762

File: packages/design-system-tailwind-preset/src/colors.test.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { colors } from './colors';
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply packages/design-system-tailwind-preset/src/colors.test.ts:1]: Keeping colors to insure initial testing and going to work on tests in a different PR https://github.com/MetaMask/metamask-design-system/issues/48

File: packages/design-system-tailwind-preset/src/colors.ts (hunk 3 above)
Comment (georgewrmarshall) [reply packages/design-system-tailwind-preset/src/colors.ts:1]: Keeping colors to insure initial testing and going to work on tests in a different PR https://github.com/MetaMask/metamask-design-system/issues/48

File: packages/design-system-tailwind-preset/README.md [hunk 6]
```diff
@@ -10,6 +10,58 @@ or
...
+- [Customizing Your Tailwind Configuration](https://tailwindcss.com/docs/configuration)
+- [Presets in Tailwind CSS](https://tailwindcss.com/docs/presets)
+
+## Contributing
+
+This package is part of a monorepo. Instructions for contributing can be found in the [monorepo README](https://github.com/MetaMask/metamask-design-system#readme).
+
 ## Contributing
 
 This package is part of a monorepo. Instructions for contributing can be found in the [monorepo README](https://github.com/MetaMask/metamask-design-system#readme).
```
Comment (garrettbear) [inline packages/design-system-tailwind-preset/README.md:67]: duplicate

File: packages/design-system-tailwind-preset/package.json [hunk 7]
```diff
@@ -51,11 +51,15 @@
...
     "jest": "^27.5.1",
+    "tailwindcss": "^3.4.13",
     "ts-jest": "^27.1.4",
     "typedoc": "^0.24.8",
     "typedoc-plugin-missing-exports": "^2.0.0",
     "typescript": "~5.2.2"
   },
+  "peerDependencies": {
+    "tailwindcss": "^3.4.13"
+  },
```
Comment (garrettbear) [inline packages/design-system-tailwind-preset/package.json:62]: For `peerDependencies`, I prefer to set the version back to `^3.0.0`. This ensures that if someone adds this package to an existing project using an earlier version (e.g., `3.3.10`), there won't be version conflicts.

File: packages/design-system-tailwind-preset/package.json [hunk 8]
```diff
@@ -51,11 +51,15 @@
...
   "engines": {
     "node": "^18.18 || >=20"
```
Comment (garrettbear) [inline packages/design-system-tailwind-preset/package.json:64]: We should consider removing the Node-specific version constraints to avoid introducing unnecessary dependency requirements into other projects.

File: packages/design-system-tailwind-preset/package.json (hunk 7 above)
Comment (georgewrmarshall) [reply packages/design-system-tailwind-preset/package.json]: Great point! Updated here https://github.com/MetaMask/metamask-design-system/pull/41/commits/9e2399491f4b98bdd3926d2a8fc715c16f0a4431

File: packages/design-system-tailwind-preset/README.md (hunk 6 above)
Comment (georgewrmarshall) [reply packages/design-system-tailwind-preset/README.md]: Great catch! Updated here https://github.com/MetaMask/metamask-design-system/pull/41/commits/9e2399491f4b98bdd3926d2a8fc715c16f0a4431

**Review by garrettbear — APPROVED** [review]

**Review by brianacnguyen — APPROVED** [review]
